Src/SemanticComparison/ProxyGenerator.cs

```diff
@@ -173,13 +183,10 @@ private static void BuildMethodGetHashCode<TDestination>(TypeBuilder type)
             method.SetReturnType(typeof(int));
 
             int derivedGetHashCode = 135;
-            MethodInfo getHashCode = typeof(TDestination).GetMethod(

```

Comment:
Yes, the `OperatingSystem` type no longer exists. I'd prefer to not create our own `OperatingSystem`. Rather, we could create some other type instead that has properties which we need (e.g. implements some specific interfaces), however we should understand that properties.

In fact I modified the single test only. From what I got it seems that we don't use any properties and the replacement is totally fine (as we override how we compare object).

It would be good if you could analyze this test change in depth and see whether my replacement is fine or not. After all, you created this particular lib 😉